src/private/useMdxComponents.tsx

```diff
@@ -74,7 +74,17 @@ export const useMdxComponents = ({ size }: Props): MDX.ProviderComponents => {
     // node is not wrapped in a paragraph and it should be, wrap it using a
     // remark plugin, not here.
     p: ({ children }) => <Text size={size}>{children}</Text>,
-    pre: ({ children }) => <pre className={styles.pre}>{children}</pre>,
+    pre: ({ children }) => (
+      <Box className={styles.pre} component="pre">
+        {children}
+      </Box>
+    ),
+    span: (props) => (
+      // For wide SVGs like Mermaid diagrams
+      <Box overflow="auto">
```

Comment:
Where else does `mdx-js` use spans?

Reply:
AFAIK not from any Markdown syntax (https://mdxjs.com/table-of-components), but a literal `<span>` element might make its way through. If we find another case we could inspect the `props`, but I'd rather punt that down the road.